src/extension/src/azure/azureServices.ts

```diff
@@ -193,6 +210,32 @@ export class AzureServices extends WizardServant {
     };
   }
 
+  public static async sendAppServiceNameValidationStatusToClient(
+    message: any
+  ): Promise<IPayloadResponse> {
+    await AzureServices.updateAppServiceSubscriptionItemCache(
+      message.subscription
+    );
+    return await AzureServices.AzureAppServiceProvider.checkWebAppName(
+      message.appName,
+      AzureServices.userAppServiceSubsctiptionItemCache
+    )
+      .then((invalidReason: string | undefined) => {
+        return {
+          payload: {
+            isAvailable:
+              !invalidReason ||
+              invalidReason === undefined ||
+              invalidReason === "",
```

Comment:
Feels like this is redundant as `!undefined` and `!""` both return true.

Reply:
I agree that it is redundant. But the logic here is exactly the same as what they have for cosmos and function. 
I was think of refactoring the Azure related code in another PR because there is a lot of duplicated code right now. 
@piotrmark What do you think?

Reply:
I think it's fine to do in a separate PR. Lot of the things we did were copied over from how Cosmos/Functions does it for consistency.

Reply:
Yes, I agree :)